nimble/controller/src/ble_ll_scan.c

```diff
@@ -1928,6 +1928,22 @@ ble_ll_scan_rx_pkt_in(uint8_t ptype, struct os_mbuf *om, struct ble_mbuf_hdr *hd
     struct ble_ll_scan_sm *scansm;
     struct ble_ll_scan_addr_data addrd;
     uint8_t *targeta;
+    uint8_t max_pdu_type;
+
+    scansm = &g_ble_ll_scan_sm;
+    rxinfo = &hdr->rxinfo;
+
+    /* Ignore PDUs we do not expect here */
+    max_pdu_type = BLE_ADV_PDU_TYPE_ADV_SCAN_IND;
```

Comment:
If I didn't miss anything, this filter is incomplete: in case extended advertisements are not enabled and an incoming packet is of type `BLE_ADV_PDU_TYPE_AUX_CONNECT_REQ`, it passes this filter (`BLE_ADV_PDU_TYPE_AUX_CONNECT_REQ := 5` which is <= `BLE_ADV_PDU_TYPE_ADV_SCAN_IND := 6`). 
The same goes for packets with type `BLE_ADV_PDU_TYPE_AUX_SCAN_REQ`...

Reply:
AUX_CONNECT_REQ and AUX_SCAN_REQ are the same PDUs as CONNECT_IND and SCAN_REQ and thus they have the same PDU type. The only difference is the channel on which they are sent - ble_ll_scan only scans on primary advertising channels.

Reply:
I see, makes sense. That is where my current knowledge about extended advertisements ends :-)